src/test/java/com/uniovi/main/services/AgentsServiceTest.java

```diff
@@ -0,0 +1,75 @@
+package com.uniovi.main.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.HttpClientErrorException;
+
+import com.uniovi.entities.AgentInfo;
+import com.uniovi.main.InciManagerI2bApplication;
+import com.uniovi.services.AgentsService;
+
+@SpringBootTest(classes = { InciManagerI2bApplication.class })
+@RunWith(SpringJUnit4ClassRunner.class)
+public class AgentsServiceTest {
+
+	@Autowired
+	private AgentsService agentsService;
+
+
+	private AgentInfo testInfo1;
+	private AgentInfo testInfo2;
+
+
+	@After
+	public void clean() {
+		if (testInfo1 != null)
+			agentsService.deleteAgent(testInfo1);
+		if (testInfo2 != null)
+			agentsService.deleteAgent(testInfo1);

```

Comment:
This is why one of the tests fail. You never really delete testInfo2 after a test method.